File: my-solutions/0069-sqrtx/solution.py
class Solution:
    def mySqrt(self, x: int) -> int:
        if (x == 0):
            return 0
        if (x == 1):
            return 1
        low = 1
        high = x
        while(low <= high):  
            mid = (low + high)//2
            
            if(mid == (x/mid)):
                return mid
            elif(mid <= (x/mid)):
                low = mid + 1
            else:
                high = mid - 1
        return high
        

File: my-solutions/0069-sqrtx/solution.java
class Solution {
    public int mySqrt(int x) {
        int start = 0;
        int end = x;
        while(start <= end){
            int mid = (end - start)/2 + start;
            long midSqrt = (long)mid * mid;
            if(midSqrt == x)
                return mid;
            if(midSqrt > x){
                end = mid - 1;
            }else{
                start = mid + 1;
            }
        }
        return end;
    }
}
